### PR TITLE
journal

### PR DESCRIPTION
**Pull request by symbiont-stevan-andjelkovic**
- feat(journal): use ns for latencies rather than ms
- feat(journal): add concurrent testing property

### CODE REVIEW

File: src/journal/test/JournalTest.hs [hunk 1]
```diff
@@ -508,9 +516,138 @@ unit_bug14 = assertProgram ""
...
+    any' _p [] = True
+    any'  p xs = any p xs
+
+prop_concurrent :: Property
+prop_concurrent = forAllConcProgram $ \(ConcProgram cmdss) -> monadicIO $ do
+  -- Rerun a couple of times, to avoid being lucky with the interleavings.
+  replicateM_ 10 $ do
+    (fp, jour) <- run initJournal
+    queue <- run newTQueueIO
+    run (mapM_ (mapConcurrently (concExec queue jour)) cmdss)
```
Comment (symbiont-daniel-gustafsson) [inline src/journal/test/JournalTest.hs:649]: Hmm this seems suspicious. Shouldn't `mapM_` and `mapConcurrently` swap places? Or maybe I'm misunderstanding the meaning of ` newtype ConcProgram = ConcProgram [[Command]]` but I thought the outer list would be different (concurrent) programs and the inner list is just a sequential program.

**Review by symbiont-daniel-gustafsson — APPROVED** [review]

File: src/journal/test/JournalTest.hs (hunk 1 above)
Comment (symbiont-stevan-andjelkovic) [reply src/journal/test/JournalTest.hs:649]: I did it like in qsm: each inner list is supposed to be executed concurrently, think of the outer list as barriers / synchronisation points between the concurrent execution.

Comment (symbiont-daniel-gustafsson) [reply src/journal/test/JournalTest.hs:649]: Hmm okay, must be me not remembering how we did it. I wonder if we could use these barriers in the linearisation check somehow..

Comment (symbiont-stevan-andjelkovic) [reply src/journal/test/JournalTest.hs:649]: Perhaps...

**Pull request merged**